### DISCUSSION TITLE
[紧急特性急需应对]b站上线新特性，分享视频链接会暴露分享者身份

### DISCUSSION THREAD

**Discussion by Slarper** [the1812/Bilibili-Evolved#3424] [功能建议 / Ideas]
这个功能是6月10日0点上线的。如图，注意网址部分（其他部分可以忽略）
![image](https://user-images.githubusercontent.com/98243861/172918496-6966f3b1-f10d-4cb6-85e5-92a67d4728ce.png)
当我们未登陆时，网址一切正常（图中情况是使用了“网址参数清理”这个组件）
但当我们已经登录b站时，再看视频网址就会自动加上一串参数`?vd_source=XXXXXXXXXXXXXXX`，如图
![image](https://user-images.githubusercontent.com/98243861/172918938-ec8224de-accf-4474-8d60-d9bb5231b91b.png)
这个`vd_source`参数经过测试，跟你的账号有关。极有可能就是你账号的uid的hash值。
这直接导致分享视频链接会暴露分享者的账号。
下图是nga今晚0点50分刚发的新帖子。可以看到这位老哥已经中招了，他的b站账号被开盒就是时间问题。（不过好像这帖是他在分享自己的战绩……）
![image](https://user-images.githubusercontent.com/98243861/172919630-1674b90b-0aa3-483c-8efc-12ac0f1d8d7d.png)
“网址参数清理”这个组件无法清理掉这串参数，甚至你手动删掉回车刷新也不行……它总是会自动刷新，给你加上这一串参数（只要你登录了b站）
这个恶心的新特性可以说不仅侵犯了用户隐私，而且也不利于收藏夹整理——它会导致浏览器不能分辨出已经收藏过的b站视频链接，从而容易重复收藏。
希望能出一个组件，不求能搞定这串参数，起码能在收藏b站视频时自动去掉一切参数，不至于污染收藏夹。

**Comment by Tinhone** [comment]
注意一下图片打码？
可能是在旧版评论区、新版评论区更新的时候一块加上去的...
奇妙的是，只要有值了它就不会再刷新，像这样：https://www.bilibili.com/video/BV1fV411W7Ss?vd_source=0
<del>第一眼还以为是什么跟踪函数</del>

**Reply by MC-dusk** [reply]
> 注意一下图片打码？ 可能是在旧版评论区、新版评论区更新的时候一块加上去的... 奇妙的是，只要有值了它就不会再刷新，像这样：https://www.bilibili.com/video/BV1fV411W7Ss?vd_source=0 ~第一眼还以为是什么跟踪函数~

我这个链接点进去跟踪参数还是会自动刷新

**Comment by the1812** [comment]
加个插件就行了 
[vdSourceClean.zip](https://github.com/the1812/Bilibili-Evolved/files/8874895/vdSourceClean.zip)

**Reply by Slarper** [reply]
完美解决~

**Reply by iseek0** [reply]
哇，厉害

**Reply by NightSharp** [reply]
这个怎么用啊

**Reply by theblueandblackcrow** [reply]
> 加个插件就行了 [vdSourceClean.zip](https://github.com/the1812/Bilibili-Evolved/files/8874895/vdSourceClean.zip)

nice啊！

**Reply by MC-dusk** [reply]
请问这个插件之后会合并到`网址参数清理 `，还是单独提供插件在线安装呢？

**Reply by the1812** [reply]
合并

**Reply by Slarper** [reply]
请问网址参数清理能不能加个判断，把最后那个反斜杠也去掉（有时会有`/`）

**Reply by Geeyun-JY3** [reply]
> 请问网址参数清理能不能加个判断，把最后那个反斜杠也去掉（有时会有`/`）

这个需求见 https://github.com/the1812/Bilibili-Evolved/issues/3053#issuecomment-1066070214

**Reply by Slarper** [reply]
thank you.

**Reply by hmgdd101** [reply]
大佬，不会用啊

**Reply by krau** [reply]
个人测试该方法无效，bilibili-evolved版本 v2.2.1-preview 。时间6.20

**Reply by Slarper** [reply]
有效呀。

**Reply by freemedom** [reply]
我试着不管用啊

**Comment by Geeyun-JY3** [comment]
这东西我昨天就发现了，点击推荐视频就能复现

**Reply by Geeyun-JY3** [reply]
不过说实话，破站客户端分享复制出来的短链接跳转后带的信息更多，我看得懂的有所用客户端平台、时间戳。
网易云音乐客户端复制的分享链接直接就直接带着版本号和 uid，都不带加密的。

**Reply by GalaxySnail** [reply]
确实，所以我每次这种短链接都是用 https://b23.wtf 转换以后再访问或者分享，如果闲的话还会用脚本把 bv 号转成 av 号（

**Reply by MC-dusk** [reply]
> 确实，所以我每次这种短链接都是用 https://b23.wtf 转换以后再访问或者分享，如果闲的话还会用脚本把 bv 号转成 av 号（

没意义啊，短链接是给别人看的，真想追踪的，手动替换回`b23.tv`跟踪信息不就看得到了吗

**Reply by GalaxySnail** [reply]
> 没意义啊，短链接是给别人看的，真想追踪的，手动替换回`b23.tv`跟踪信息不就看得到了吗

你理解错了，我的意思是发送 b23.wtf 重定向之后的链接，当然不包括原始的短链接

**Comment by Slarper** [comment]
虚惊一场哈哈

**Comment by freemedom** [comment]
![image](https://user-images.githubusercontent.com/57294686/179335626-f61dd746-9089-48a1-a8cc-e36a1b16f5bc.png)
刚刚测试了一下，不管用啊

**Reply by freemedom** [reply]
知道了，安成插件了，安成组件就可以了

**Reply by PalmerCharles** [reply]
请问图中的是什么扩展插件，脚本猫吗

**Reply by Geeyun-JY3** [reply]
> 请问图中的是什么扩展插件，脚本猫吗

这就是本脚本的设置面板啊

**Comment by freemedom** [comment]
![image](https://user-images.githubusercontent.com/57294686/179336633-abd00205-0819-43e4-9195-6c6869d4f69f.png)

**Comment by freemedom** [comment]
let c = document.URL.replace(window.location.search, "");
window.history.replaceState(history.state, "", c)

**Comment by howdareyo** [comment]
因为这是js脚本加载导致的
https://s1.hdslb.com/bfs/static/jinkela/video/stardust-video.309d6bdd251542f3a178dd340bc753e2d7ebfcf1.js
屏蔽这个脚本会导致评论区,合集,up头像加载不出来
vd_source就是用户信息的hash啊
等大佬8

**Reply by MC-dusk** [reply]
早就解决了
https://github.com/the1812/Bilibili-Evolved/discussions/3424#discussioncomment-2918252

**Comment by arenekosreal** [comment]
https://greasyfork.org/scripts/393995
目前看来这个脚本可破，能否合并进参数清理插件中？

**Reply by MC-dusk** [reply]
早都解决了
https://github.com/the1812/Bilibili-Evolved/discussions/3424#discussioncomment-2918252

**Comment by Ethkuil** [comment]
这个问题已经解决了吧？谁来 `Mark as answer` 一下？

**Reply by the1812** [reply]
这是功能建议，已经标记为已实现了

**Reply by Ethkuil** [reply]
贴下对应哪个功能、然后标记为答案，这样可能会更好，方便后来的用户快速知道功能已实现以及在哪启用。（比如7月就加了的功能，上面就有好几位到8月、9月了还以为没实现[doge]

**Reply by the1812** [reply]
只有提问类的可以标记答案...
启用的话就是打开 `网址参数清理`, 保持更新就行了

**Reply by Geeyun-JY3** [reply]
打上标签其实就能知道做完没了。目前也一直有在这么做。

要知道对应哪个功能这个，确实不怎么方便找对应 commit，这和 GitHub Discussion 的机制有点关系。
只能建议用户学会在看到已被打上已完成相关标签后善用 GitHub 搜索在 commit 里搜对应 discussion 的 id 了。

GitHub Discussion 好像就是故意地不搞和 issue、PR、commit 之间的交叉引用互相显示
GitHub 官方讨论区有人提了这事也挺久了：

- https://github.com/orgs/community/discussions/2971
- https://github.com/orgs/community/discussions/3562

**Comment by TomoeMami** [comment]
经过测试，目前ClearURLs插件已经可以正确识别并处理掉vd_source参数
